### DISCUSSION TITLE
Elsa 3.1 - How to integrating elsa 3.1 with sql server ?

### DISCUSSION THREAD

**Discussion by nurhidayat-agung** [elsa-workflows/elsa-core#5576] [Q&A]
hello elsa ,
how to integrating elsa 3.0 with sql server to replace sqlite ?
what package idi i need to install ?
how to perform manual migration ?
is db contenst create automatically or we mus create manual one ?

**Comment by glime-ai** [comment]
To integrate Elsa 3.0 with SQL Server instead of SQLite, you will need to install the appropriate NuGet package for Entity Framework Core that supports SQL Server. As of my knowledge cutoff date in early 2023, the specific package for Elsa 3.0 and SQL Server integration may not be directly documented. However, you can follow the general guidelines for configuring Entity Framework Core with SQL Server and apply them to Elsa.

Here are the steps you would typically follow:

1. **Install the NuGet package**: You need to install the `Elsa.Persistence.EntityFramework.SqlServer` package. This can be done using the following command in your project directory:

   ```
   dotnet add package Elsa.Persistence.EntityFramework.SqlServer
   ```

2. **Configure services**: In your `Program.cs` or `Startup.cs` (depending on your project setup), you need to configure the services to use SQL Server. Here's an example of how you might do it:

   ```csharp
   builder.Services.AddElsa(elsa =>
   {
       elsa.UseEntityFrameworkPersistence(ef => ef.UseSqlServer("Your SQL Server connection string"));
   });
   ```

3. **Perform manual migrations**: To create the necessary database schema, you will need to perform migrations. You can do this using the Entity Framework Core tools. First, ensure you have the `Microsoft.EntityFrameworkCore.Design` package installed, then run the following commands:

   ```
   dotnet ef migrations add InitialCreate --context ElsaContext
   dotnet ef database update --context ElsaContext
   ```

   Replace `ElsaContext` with the actual name of your DbContext if it's different.

4. **Database content**: The migrations will create the necessary tables and schema in your SQL Server database. You do not need to create these manually if the migrations are successful.

Please note that the exact package names and commands may vary depending on the version of Elsa and Entity Framework Core you are using. Always refer to the official documentation or the package source for the most accurate and up-to-date information.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
